libraries/chain/include/eosio/chain/block_header_state.hpp

```diff
@@ -45,46 +54,57 @@ struct pending_block_header_state {
  *  @struct block_header_state
  *  @brief defines the minimum state necessary to validate transaction headers
  */
-struct block_header_state {
-    block_id_type                     id;
-    uint32_t                          block_num = 0;
-    signed_block_header               header;
-    uint32_t                          dpos_proposed_irreversible_blocknum = 0;
-    uint32_t                          dpos_irreversible_blocknum = 0;
-    uint32_t                          bft_irreversible_blocknum = 0;
-    uint32_t                          pending_schedule_lib_num = 0; /// last irr block num
-    digest_type                       pending_schedule_hash;
-    producer_schedule_type            pending_schedule;
-    producer_schedule_type            active_schedule;
-    incremental_merkle                blockroot_merkle;
-    flat_map<account_name,uint32_t>   producer_to_last_produced;
-    flat_map<account_name,uint32_t>   producer_to_last_implied_irb;
-    public_key_type                   block_signing_key;
-    vector<uint8_t>                   confirm_count;
+struct block_header_state : public detail::block_header_state_common {
+   block_id_type                     id;
+   signed_block_header               header;
+   detail::schedule_info             pending_schedule;
+
+   block_header_state() = default;
+
+   block_header_state( detail::block_header_state_common&& base )
+   :detail::block_header_state_common( std::move(base) )
+   {}
```

Comment:
explicit constructor here would provide guard against accident creation from block_header_state_common.

Reply:
Resolved in https://github.com/EOSIO/eos/pull/6624/files#diff-3701e88f728b2bd0a37f5d347a042124R64.